server/src/main/kotlin/suwayomi/tachidesk/manga/impl/update/Updater.kt

```diff
@@ -0,0 +1,72 @@
+package suwayomi.tachidesk.manga.impl.update
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import mu.KotlinLogging
+import suwayomi.tachidesk.manga.impl.Chapter
+import suwayomi.tachidesk.manga.model.dataclass.MangaDataClass
+
+class Updater : IUpdater {
+    private val logger = KotlinLogging.logger {}
+    private val scope = CoroutineScope(SupervisorJob() + Dispatchers.Default)
+
+    private var isRunning: Boolean = false
```

Comment:
Maybe check if channel.isActive or updateJob.isActive instead?

Reply:
don't actually need isRunning so i'll remove it completely